### ISSUE TITLE
Rule CanBeNonNullable should not report when function is overridden

### ISSUE THREAD

**Issue by marschwar** [detekt/detekt#4486]
## Expected Behavior
The following code should not trigger the `CanBeNonNullable` rule as the implementing class cannot change the signature of the function from the interface.

```kotlin
interface I {
    fun foo(bar: Int?) : Int
}

class C : I {
    override fun foo(bar: Int?) : Int {
        return bar!!
    }
}
```
## Observed Behavior
The code is reported with the message: _The nullable parameter 'bar' can be made non-nullable._ at Location(source=6:22, ...)
